ctypesgen/__main__.py

```diff
@@ -342,82 +295,76 @@ def main(givenargs=None):
     )
 
     # Error options
-    op.add_option(
-        "",
+    parser.add_argument(
         "--all-errors",
         action="store_true",
         default=False,
         dest="show_all_errors",
-        help="Display all warnings and errors even " "if they would not affect output.",
+        help="Display all warnings and errors even if they would not affect output.",
     )
-    op.add_option(
-        "",
+    parser.add_argument(
         "--show-long-errors",
         action="store_true",
         default=False,
         dest="show_long_errors",
-        help="Display long error messages " "instead of abbreviating error messages.",
+        help="Display long error messages instead of abbreviating error messages.",
     )
-    op.add_option(
-        "",
+    parser.add_argument(
         "--no-macro-warnings",
         action="store_false",
         default=True,
         dest="show_macro_warnings",
         help="Do not print macro warnings.",
     )
-    op.add_option(
-        "",
+    parser.add_argument(
         "--debug-level",
         dest="debug_level",
         default=0,
-        type="int",
+        type=int,
         help="Run ctypesgen with specified debug level (also applies to yacc parser)",
     )
 
-    op.set_defaults(**core_options.default_values)
+    parser.set_defaults(**core_options.default_values)
+    args = parser.parse_args(givenargs)
 
-    (options, args) = op.parse_args(givenargs)
-    options.headers = args
+    args.compile_libdirs += args.universal_libdirs
+    args.runtime_libdirs += args.universal_libdirs
```

Comment:
Turns out python list `+=` actually mutates the object on the left (instead of creating a copy), which can be problematic when ctypesgen is called repeatedly through the API: If the default `[]` is used, it is modified, so the next run's default compile/runtime libdirs would still contain the previous run's universal libdirs.
I think this may be happening in the test suite, but has so far passed unnoticed, as specifying too many libdirs (or redundantly) usually doesn't cause any immediate issues.

Python's behavior of `a += b` modifying while `a = a + b` does not feels slightly confusing, though. Normally you'd expect both syntaxes to do the same thing.

@nilason You might want to patch this at some point.